pkg/store/ristretto_cache.go

```diff
@@ -0,0 +1,39 @@
+package store
+
+import (
+	"github.com/dgraph-io/ristretto"
+	"github.com/finitum/AAAAA/pkg/aur"
+	"github.com/pkg/errors"
+)
+
+type Ristretto struct {
+	cache *ristretto.Cache
+}
+
+func NewRistretto() (*Ristretto, error) {
+	// TODO: these are sane defaults, we may want to make them configurable
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 1e7,
+		MaxCost:     1 << 30,
+		BufferItems: 64,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "creating ristretto cache")
+	}
+
+	return &Ristretto{cache}, nil
+}
+
+func (r Ristretto) SetEntry(term string, result aur.Results) error {
+	r.cache.SetWithTTL(term, result, int64(len(result)), cacheTTL)
```

Comment:
TTL is optional for ristretto as it also can manage the cache based on the MaxCost, not sure if we therefore want to keep the ttl for ristretto or not